File: 268_Missing_Number_A2.c
#include<stdio.h>
#include<stdlib.h>

int missingNumber(int* nums, int numsSize) {
    int all_sum = numsSize * (numsSize + 1) / 2;
    int sum = 0;

    for (int i = 0; i < numsSize; i++) {
        sum += nums[i];
    }

    return all_sum - sum;
}

int main() {
    int nums[] = {3, 0, 1, 2};
    int numsSize = sizeof(nums) / sizeof(nums[0]);

    int missing = missingNumber(nums, numsSize);
    printf("Missing number: %d\n", missing);

    return 0;
}

File: 268_Missing_Number_A2.py
class Solution:
    def missingNumber(self, nums: list[int]) -> int:
        all_sum = sum(range(len(nums)+1))
        # all_sum = (len(nums) * (len(nums) + 1)) // 2
        nums_sum = sum(nums)

        return all_sum - nums_sum
    

a = Solution()
print(a.missingNumber([3,0,1])) # 2
print(a.missingNumber([0,1])) # 2
print(a.missingNumber([9,6,4,2,3,5,7,0,1])) # 8